crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -81,12 +79,14 @@ impl Object {
     async fn previous_transaction_block(
         &self,
         ctx: &Context<'_>,
-    ) -> Result<Option<TransactionBlock>> {
-        if let Some(tx) = &self.previous_transaction {
-            let loader = ctx.data_unchecked::<DataLoader<SuiClientLoader, LruCache>>();
-            Ok(loader.load_one(*tx).await.unwrap_or(None))
-        } else {
-            Ok(None)
+    ) -> Result<Option<TransactionBlock>, crate::error::Error> {
+        match self.previous_transaction {
+            Some(digest) => {
```

Comment:
ah oops, never cleaned this up, this is not ideal since fetch_tx takes the &str and creates Digest::from_str(digest)?.into_vec(). It seems like we're better off just passing in &Digest directly. Not blocking though.

Reply:
Good point, let's do this in a follow up PR.